plugins/router/f5/f5.go

```diff
@@ -0,0 +1,555 @@
+	}
+
+	// We don't really need to initialise f5.poolMembers[poolName] because
+	// we always check whether it is initialised before using it, but
+	// initialising it to an empty map here saves a REST call later the first
+	// time f5.PoolHasMember is invoked with poolName.
+	f5.poolMembers[poolName] = map[string]bool{}
+
+	glog.V(4).Infof("Pool %s created.", poolName)
+
+	return nil
+}
+
+// Delete the pool named poolName from F5, and delete
+// f5.poolMembers[poolName].
+func (f5 *f5LTM) DeletePool(poolName string) error {
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool/%s", f5.host, poolName)
+
+	err := f5.delete(url, nil)
+	if err != nil {
+		return err
+	}
+
+	delete(f5.poolMembers, poolName)
+
+	glog.V(4).Infof("Pool %s deleted.", poolName)
+
+	return nil
+}
+
+type f5PoolMember struct {
+	Name string `json:"name"`
+}
+
+type f5Pool struct {
+	Members []f5PoolMember `json:"items"`
+	Code    int            `json:"code"`
+	Message string         `json:"message"`
+}
+
+// Return f5.poolMembers[poolName], first initializing it from F5 if it is zero.
+func (f5 *f5LTM) GetPoolMembers(poolName string) (map[string]bool, error) {
+	members, ok := f5.poolMembers[poolName]
+	if ok {
+		return members, nil
+	}
+
+	url := fmt.Sprintf("https://%s/mgmt/tm/ltm/pool/%s/members",
+		f5.host, poolName)
+
+	res := struct {
+		Members []f5PoolMember `json:"items"`
+		Code    int            `json:"code"`
+		Message string         `json:"message"`
+	}{}
+
+	err := f5.get(url, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	// Note that we do not initialise f5.poolMembers[poolName] unless we know that
+	// the pool exists (i.e., the above GET request for the pool succeeds).
+	f5.poolMembers[poolName] = map[string]bool{}
+
+	for _, member := range res.Members {
+		f5.poolMembers[poolName][member.Name] = true
+	}
+
+	return f5.poolMembers[poolName], nil
+}
+
+// Check whether the specified pool exists.  Internally, this method uses
+// f5.poolMembers[poolName], as a side effect initialising it if it is zero.
+func (f5 *f5LTM) PoolExists(poolName string) (bool, error) {
+	_, err := f5.GetPoolMembers(poolName)
+	if err == nil {
+		return true, nil
+	}
+
+	if err.(F5Error).httpStatusCode == 404 {
+		return false, nil
+	}
+
+	return false, err
+}
+
+// Check whether the given member is in the pool named poolName on F5.
+// Internally, this method uses f5.poolMembers[poolName], initialising
+// it first if it is zero.
+func (f5 *f5LTM) PoolHasMember(poolName string, member string) (bool, error) {
+	members, err := f5.GetPoolMembers(poolName)
+	if err != nil {
+		return false, err
+	}
+
+	return members[member], nil
+}
+
+// Add member to the pool named poolName on F5, and update
+// f5.poolMembers[poolName].
+func (f5 *f5LTM) AddPoolMember(poolName string, member string) error {
+	hasMember, err := f5.PoolHasMember(poolName, member)
+	if err != nil {
+		return err
+	}
+	if hasMember {
+		return fmt.Errorf("Pool %s already has member %s.\n", poolName, member)
```

Comment:
The plug-in should check whether the member is in the pool before it tries to add it.  There is a logic error if we hit this case.  If you believe it would be better to log and continue though, I can make that change.